llmfoundry/models/mpt/modeling_mpt.py

```diff
@@ -1015,9 +1017,18 @@ def forward(self, batch: MutableMapping) -> CausalLMOutputWithPast:
 
     def loss(self, outputs: CausalLMOutputWithPast,
              batch: Mapping) -> torch.Tensor:
-        targets = self.get_targets(batch)
-        return self.loss_fn(outputs.logits.view(-1, outputs.logits.size(-1)),
-                            targets.view(-1))
+        targets = self.get_targets(batch).view(-1)
```

Comment:
The only place targets is used is in `(targets != self.loss_fn.ignore_index).sum()`.

Do we need to view it as -1 for that?

(I'm a bit scared of destroying batch dim unless its NEEDED)

Reply:
There is only one other place the targets is being used in this function, and that also uses `.view(1)`

<img width="1337" alt="Screenshot 2024-04-01 at 5 12 21 PM" src="https://github.com/mosaicml/llm-foundry/assets/144760128/13850946-3b84-4419-b24b-7f22f5115e42">

Reply:
@vchiley CE loss expects the flattened targets, right?

Reply:
Just trying to future proof the code.
I'd argue that we should pass it into the loss fn as `targets.view(1)` but don't destroy the batch info unless needed.
(what if in the future someone wants the batch info for something else)

Reply:
i guess ppl can undo this change in the future, I'm just not sure what the point of including it now is 🤷‍♂️

Reply:
Makes sense, let me update the PR.